Running tox test suite seems to require `export LANG=C` for `tests/test_ext_autodoc.py::test_parse_name` but not when individually tested

### Describe the bug

Issues with testing on my locale, `export LANG=C` needed for `tox`.



### How to Reproduce

```
$ git clone https://github.com/sphinx-doc/sphinx.git
```

then (tested at 873d9f6fd)

```
$ tox -e py310 tests/test_ext_autodoc.py
```

completes with no failure reported
```
  py310: commands succeeded
  congratulations :)
```

but 

```
tox -e py310
```

complains

```
FAILED tests/test_ext_autodoc.py::test_parse_name - AssertionError: assert 's...
===== 1 failed, 1785 passed, 18 skipped, 10 warnings in 438.70s (0:07:18) ======
ERROR: InvocationError for command /path/to/sphinx/.tox/py310/bin/python -X dev -X warn_default_encoding -m pytest --durations 25 (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py310: commands failed
```

On the other hand

```
export LANG=C
tox -e py310
```

completes with no complain.

But `export LANG=C` is not needed for individual testing:

```
$ echo $LANG
fr_FR.UTF-8
$ tox -e py310 tests/test_ext_autodoc.py::test_parse_name
py310 develop-inst-noop: /path/to/sphinx
....
py310 run-test: commands[0] | python -X dev -X warn_default_encoding -m pytest --durations 25 tests/test_ext_autodoc.py::test_parse_name
/path/to/sphinx/.tox/py310/lib/python3.10/site-packages/iniconfig/__init__.py:52: EncodingWarning: 'encoding' argument not specified
  f = open(self.path)
...
============================= test session starts ==============================
platform darwin -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0
cachedir: .tox/py310/.pytest_cache
libraries: Sphinx-5.1.0+/873d9f6fd, docutils-0.18.1
base tempdir: /private/var/folders/np/4_9pppfs54xdv9m79m7rwx4w0000gn/T/pytest-of-jfb/pytest-13
rootdir: /path/to/sphinx, configfile: setup.cfg
collected 1 item                                                               

tests/test_ext_autodoc.py .                                              [100%]

=============================== warnings summary ===============================
.tox/py310/lib/python3.10/site-packages/_pytest/capture.py:357
  /path/to/sphinx/.tox/py310/lib/python3.10/site-packages/_pytest/capture.py:357: EncodingWarning: 'encoding' argument not specified
    self.tmpfile = open(os.devnull)

.tox/py310/lib/python3.10/site-packages/_pytest/cacheprovider.py:189
.tox/py310/lib/python3.10/site-packages/_pytest/cacheprovider.py:189
.tox/py310/lib/python3.10/site-packages/_pytest/cacheprovider.py:189
  /path/to/sphinx/.tox/py310/lib/python3.10/site-packages/_pytest/cacheprovider.py:189: EncodingWarning: 'encoding' argument not specified
    f = path.open("w")

.tox/py310/lib/python3.10/site-packages/_pytest/cacheprovider.py:160
  /path/to/sphinx/.tox/py310/lib/python3.10/site-packages/_pytest/cacheprovider.py:160: EncodingWarning: 'encoding' argument not specified
    with path.open("r") as f:

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================= slowest 25 durations =============================
0.26s setup    tests/test_ext_autodoc.py::test_parse_name

(2 durations < 0.005s hidden.  Use -vv to show these durations.)
======================== 1 passed, 5 warnings in 0.35s =========================
___________________________________ summary ____________________________________
  py310: commands succeeded
  congratulations :)
```


### Expected behavior

Global test suite should not differ from individual test.

### Your project

Sphinx

### OS

mac os

### Python version

3.10.4

### Sphinx version

5.x/873d9f6fd

Actually I observed this since a long time ago.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -5,8 +5,20 @@ import docutils
 import pytest
 
 import sphinx
+import sphinx.locale
 from sphinx.testing import comparer
 from sphinx.testing.path import path
+
+def _init_console(locale_dir=sphinx.locale._LOCALE_DIR, catalog='sphinx'):
+    """Monkeypatch ``init_console`` to skip its action.
+
+    Some tests rely on warning messages in English. We don't want
+    CLI tests to bleed over those tests and make their warnings
+    translated.
+    """
+    return sphinx.locale.NullTranslations(), False
+
+sphinx.locale.init_console = _init_console
 
 pytest_plugins = 'sphinx.testing.fixtures'
 

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -19,6 +19,7 @@ def _init_console(locale_dir=sphinx.locale._LOCALE_DIR, catalog='sphinx'):
     """
     return sphinx.locale.NullTranslations(), False
 
+
 sphinx.locale.init_console = _init_console
 
 pytest_plugins = 'sphinx.testing.fixtures'

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -9,6 +9,7 @@ import sphinx.locale
 from sphinx.testing import comparer
 from sphinx.testing.path import path
 
+
 def _init_console(locale_dir=sphinx.locale._LOCALE_DIR, catalog='sphinx'):
     """Monkeypatch ``init_console`` to skip its action.
 

